sdk/cosmos/azure-cosmos/azure/cosmos/aio/_cosmos_client_connection_async.py

```diff
@@ -2349,6 +2351,47 @@ def __GetBodiesFromQueryResult(result):
         # Query operations will use ReadEndpoint even though it uses POST(for regular query operations)
         request_params = _request_object.RequestObject(typ, documents._OperationType.SqlQuery)
         req_headers = base.GetHeaders(self, initial_headers, "post", path, id_, typ, options, partition_key_range_id)
+
+        # check if query has prefix partition key
+        cont_prop = kwargs.pop("containerProperties", None)
+        partition_key = req_headers.get(http_constants.HttpHeaders.PartitionKey)
+        isPrefixPartitionQuery = False
+        partition_key_definition = None
+        if cont_prop:
+            cont_prop = await cont_prop()
+            pk_properties = cont_prop["partitionKey"]
+            partition_key_definition = PartitionKey(path=pk_properties["paths"], kind=pk_properties["kind"])
+            if partition_key_definition.kind != "MultiHash":
+                isPrefixPartitionQuery = False
+            elif type(partition_key) == list and len(partition_key_definition['paths']) == len(partition_key):
+                isPrefixPartitionQuery = False
+            else:
+                isPrefixPartitionQuery = True
+
+        if isPrefixPartitionQuery:
+            # here get the overlap, then do one of two scenarios
+            req_headers.pop(http_constants.HttpHeaders.PartitionKey, None)
+            partition_key = json.loads(partition_key)
+            feedrangeEPK = partition_key_definition.get_epk_range_for_prefix_partition_key(partition_key)
+            over_lapping_ranges = await self._routing_map_provider.get_overlapping_ranges(id_, [feedrangeEPK])
+
+            if over_lapping_ranges:
```

Comment:
Again, what about the split partition scenario? Also is there a way to have this handling be done in `_cosmos_client_connection.py` so there is not a repeat of the same logic?

Reply:
As mentioned before the split partition scenario is addressed in the routing map provider.